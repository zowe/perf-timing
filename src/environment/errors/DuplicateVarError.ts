/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

/**
 * This error represents the condition where an environment variable was
 * registered twice, indicating a conflict.
 *
 * @example
 * ```typescript
 * Environment.register("key1", "val");
 * Environment.register("key2", "val2"); // Throws error
 * ```
 */
export class DuplicateVarError extends Error {
    /**
     * Construct the error.
     * @param key The environment key in conflict
     * @param value The current default value that was set
     * @param newValue The new default value
     */
    constructor(key: string, value: any, newValue: any) {
        super(`"${key}" was previously registered with ${value.toString()}. Unable to re-register it with ${newValue.toString()}`);
    }
}
