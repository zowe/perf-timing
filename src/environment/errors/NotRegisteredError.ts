export class NotRegisteredError extends Error {
    constructor(name: string) {
        super(`The environment variable "${name}" has not been registered! It must be registered before the value can be retrieved.`);
    }
}
