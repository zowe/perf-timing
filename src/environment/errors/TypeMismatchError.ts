/*!
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

 /**
  * This error represents the condition when a getter function of the {@link Environment},
  * like {@link EnvironmentManager.getNumber}, is called and the type of the default didn't match
  * the method return value.
  *
  * @example
  * Environment.register("TEST", "string");
  * Environment.getNumber("TEST"); // Throws TypeMismatchError
  */
export class TypeMismatchError extends Error {
    /**
     * Construct the error.
     * @param key The key in error.
     * @param expectedType The expected type. Usually determined the type of call expected (getNumber, getString, etc).
     * @param actualType The actual type. Usually determined by the `typeof` the registered default.
     */
    constructor(public key: string, public expectedType: string, public actualType: string) {
        super(`Expected "${key}" to be of type "${expectedType}", instead it was "${actualType}"`);
    }
}
