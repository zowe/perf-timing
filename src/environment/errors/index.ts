export * from "./DuplicateVarError";
export * from "./NotRegisteredError";