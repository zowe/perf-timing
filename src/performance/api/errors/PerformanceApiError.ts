/*!
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

 /**
  * Base error for issues thrown by the Performance API.
  *
  * @external
  */
export class PerformanceApiError extends Error {
    /**
     * Construct the error.
     * @param message Message sent by the child.
     */
    constructor(message: string) {
        super(`Performance Error\n${message}`);
    }
}
