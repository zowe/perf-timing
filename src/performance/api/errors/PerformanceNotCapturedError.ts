/*!
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import { PerformanceApiError } from "./PerformanceApiError";

/**
 * Thrown when metric output methods were called and performance was not enabled.
 *
 * @example
 * const perfApi = new PerformanceApi({enabled: false, packageUUID: "bogus"});
 *
 * // Calling these methods will cause this error to be thrown.
 * perfApi.getMetrics();
 * perfApi.getNodeTiming();
 * perfApi.getSysInfo();
 */
export class PerformanceNotCapturedError extends PerformanceApiError {
    constructor() {
        super("Unable to access performance metrics when performance monitoring is not enabled!");
    }
}
