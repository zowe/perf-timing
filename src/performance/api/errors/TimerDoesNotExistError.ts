/*!
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

/*!
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

/*
 * This program and the accompanying materials are made available under the terms of the *
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at  *
 * https://www.eclipse.org/legal/epl-v20.html                                            *
 *                                                                                       *
 * SPDX-License-Identifier: EPL-2.0                                                      *
 *                                                                                       *
 * Copyright Contributors to the Zowe Project.                                           *
 *                                                                                       *
 */

import { PerformanceApiError } from "./PerformanceApiError";

/**
 * @TODO better documentation
 *
 * This error is thrown when a function timer is attempted to be closed but no
 * timer exists currently.
 */
export class TimerDoesNotExistError extends PerformanceApiError {
    constructor(name: string) {
        super(`A function timer with the name "${name}", does not exist. Please create the timer first.`);
    }
}
