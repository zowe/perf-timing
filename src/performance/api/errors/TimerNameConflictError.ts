/*!
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */


import { PerformanceApiError } from "./PerformanceApiError";

/**
 * @TODO BETTER DOCUMENT
 *
 * Error thrown when a timer in the functionTimers map already exists.
 */
export class TimerNameConflictError extends PerformanceApiError {
    constructor(name: string) {
        super(`A timer with the name "${name} was previously created. Please create unique timer names.`);
    }
}
