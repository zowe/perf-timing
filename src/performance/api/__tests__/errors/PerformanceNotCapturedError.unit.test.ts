/*!
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import { PerformanceApiError, PerformanceNotCapturedError } from "../../errors";

describe("PerformanceNotCapturedError", () => {
    it("should remain unchanged", () => {
        const error = new PerformanceNotCapturedError();

        expect(error).toBeInstanceOf(PerformanceApiError);
        expect(error).toMatchSnapshot();
    });
});
