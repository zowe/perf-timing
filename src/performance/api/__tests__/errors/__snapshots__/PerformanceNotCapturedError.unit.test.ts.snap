// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerformanceNotCapturedError should remain unchanged 1`] = `
[Error: Performance Error
Unable to access performance metrics when performance monitoring is not enabled!]
`;
